ddtrace/tracer/tracer.go

```diff
@@ -171,7 +182,7 @@ func newTracer(opts ...StartOption) *tracer {
 		defer t.wg.Done()
 		t.reportHealthMetrics(statsInterval)
 	}()
-	return t
+	return t, info
```

Comment:
What's the point of having this new return values? Let's get rid of it. We can simply call something like `logStartup(t)` inside `Start` and that's it. Am I missing something?

Reply:
The reason is that there is info that is not held in the tracer struct (rules sampler parsing errors, for instance) that we want to report. I started adding fields to the tracer struct initially, but that seemed wrong. I don't think we want to use the tracer struct to pass startup diagnostics between `newUnstartedTracer` and `Start`. 

Instead, I opted to return a separate structure, just for startup info.

This isn't my favorite. If you can think of a better way, I would appreciate input.

Reply:
> The reason is that there is info that is not held in the tracer struct (rules sampler parsing errors, for instance)

Can't we just call `samplingRulesFromEnv()` again and get the error? We can completely drop `newStartupInfo` and just do `logStartup(t)`. In tests we could then use the test logger and assert the output.

Reply:
That sounds like a good plan. I'll do that. It makes this much cleaner, with fewer files changed. :)